### PR TITLE
Different way of assigning Seq Nums for Raw Seed Items

### PR DESCRIPTION
**Pull request by stkem**
Instead of being a separate plugins it’s now done synchronously as part
of data ingestions, which is basically the only part that writes to
this table. This should stop the db lock contention issues.
Not a big change, but hopefully the right one.

@yingjieMiao @eishay

### CODE REVIEW

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala [hunk 1]
```diff
@@ -50,8 +51,11 @@ class SeedIngestionCommander @Inject() (
 
       val ingestKeepsF = ingestAllKeeps().flatMap { _ =>
         log.info("XYZ: ingest all keeps future completed")
+        rawSeedSeqNumAssigner.assignSequenceNumbers()
         val userIds = usersToIngestGraphDataFor()
-        FutureHelpers.sequentialExec(userIds)(ingestTopUris)
+        val resFut = FutureHelpers.sequentialExec(userIds)(ingestTopUris)
+        rawSeedSeqNumAssigner.assignSequenceNumbers()
```
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala:57]: Do you need to wait for `resFut` completes, then call `assignSeq`?

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala [hunk 2]
```diff
@@ -50,8 +51,11 @@ class SeedIngestionCommander @Inject() (
 
       val ingestKeepsF = ingestAllKeeps().flatMap { _ =>
         log.info("XYZ: ingest all keeps future completed")
+        rawSeedSeqNumAssigner.assignSequenceNumbers()
```
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala:54]: why call this twice?

Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala:54]: to reduce latency. The graph ingestion below can take quite a while and I want new keeps to get a sequence number quickly.

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala (hunk 1 above)
Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala:57]: I do indeed! Good find. Fixing it now.

**Comment by yingjieMiao** [comment]
Let's see how it works. 

As I commented in an earlier PR, an alternative is to gather scored items for 8 (or whatever `locksize`) users concurrently, and persist all data in 1 thread. In that case, the db sessions on that table is roughly reduced from `8 + 1` to `1 + 1`, 1 being the seq assigner. 

I am not fully sure about my theory though.

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala (hunk 2 above)
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala:54]: ah, I see, it's actually for the previous step. Makes sense.

**Pull request merged**